### ISSUE TITLE
[Question] How did PoP mining get fixed after BTC testnet difficulty exploit?

### ISSUE THREAD

**Issue by a123b** [VeriBlock/nodecore-releases#89]
The last known Bitcoin block of the testnet blockchain is now over 14000 blocks away from the current Bitcoin testnet block, which is steadily increasing. For a PoP transaction to be valid it would have to include context headers for all of these blocks, which makes the transaction so big that it would exceed the maximum VBK block size.

This means there is no way to ever create a valid block containing a valid PoP transaction which increases the last known Bitcoin height. It seems like an update to both nodecore (to allow such a huge PoP transaction, or make it possible to spread it over multiple VBK blocks) and the PoP miner (it only keeps track of the last 5000 blocks in its spvchain) would be required to make PoP work again.

**Comment by a123b** [comment]
Okay, never mind, PoP is working again. Seems like I misunderstood something about how PoP works :sweat_smile:

**Comment by a123b** [comment]
Okay, I'm really confused about this...
Block 62111, which solved the issue, contained 14461 Bitcoin block headers, so it seems I got right that all these headers need to be in the transaction to be valid, but it seems I was wrong about them being too big to be included in a single VBK block.
But then, why didn't I get a single PoP transaction included in a block the whole night? I self-compiled bitcoinj after changing the spvchain capacity to 30.000 so that my PoP miner can actually include that many blocks in the transaction, which it did, but the transaction never got included in a block.
So why did an even bigger PoP transaction suddenly get included in a block now? I would really appreciate if someone could explain it. Not that it matters much, but it is really boggling my mind :confused:

**Comment by JoshRack** [comment]
Apologies for not answering sooner...I can't remember entirely how this was resolved, but inasmuch as I remember:

1. You are correct in that the VeriBlock blockchain maintains something of an SPV view of the Bitcoin blockchain. New blocks are revealed to it by PoP transactions Block of Proof and context blocks. That said, the context blocks are the bridge between the block of proof and the known SPV view.
2. The very first iterations of the PoP miner required running a full Bitcoin daemon and interacted via Bitcoin's RPC API. We dusted off an old version of that miner to allow us to build a PoP transaction with the required number of context blocks (14,461 as you documented) to properly connect.
3. This is the haziest part in my memory. While such a large PoP transaction does not break any consensus rules of the blockchain, it was not preferred by the pool software. I believe a small tweak was temporarily made to a pool so that it would be selected and wind up in a block.

I do want to remind everyone that this difficulty bomb and what ensued in the VeriBlock testnet is a nuance of Bitcoin's testnet and is not something that would arise on mainnet.
